src/devices/video/mc6845.cpp

```diff
@@ -406,6 +406,11 @@ int mc6845_device::vsync_r()
 
 void mc6845_device::recompute_parameters(bool postload)
 {
+	bool zero_horizontal_width = (m_horiz_disp == 0);
+	bool zero_vertical_height = (m_vert_disp == 0);
+	bool maintain_visible_area = zero_horizontal_width || zero_vertical_height;
+	uint8_t visual_adjustment = maintain_visible_area ? 0 : 1;
```

Comment:
You’ll still need to inset the visible area rectangle if you’re maintaining the visible area since your width/height are exclusive while `rectangle` coordinates are inclusive – you don’t need `visual_adjustment`.